azure-proto-compute/AvailabilitySetContainer.cs

```diff
@@ -12,80 +12,86 @@ namespace azure_proto_compute
     /// <summary>
     /// Operatiosn class for Availability Set Contaienrs (resource groups)
     /// </summary>
-    public class AvailabilitySetContainer : ResourceContainerOperations<XAvailabilitySet, PhAvailabilitySet>
+    public class AvailabilitySetContainer : ResourceContainerOperations<AvailabilitySet, AvailabilitySetData>
     {
-        public AvailabilitySetContainer(ArmClientContext context, PhResourceGroup resourceGroup, ArmClientOptions clientOptions) : base(context, resourceGroup, clientOptions) { }
+        internal AvailabilitySetContainer(ArmClientContext context, ResourceGroupData resourceGroup, ArmClientOptions clientOptions)
```

Comment:
I wonder if, given the return types from most of our methods, this should have constructors that accept ResourceGroup and ResourceGroupOperations.  Also, I assume we are combining client context and client options in a subsequent PR

Reply:
context / options is in a future work item.

Another future item is to remove the constructors on Operations classes that take the full resource.  The idea is if you have a full resource you construct a VirtualMachine, if you have just a ResourceIdentifier you construct a VirtualMachineOperations (both internally of course)